# Introduction

The consumer experience is vital to companies, and words can be very powerful when determining the fate of a product. Our group wanted to see how effectively products on Amazon with hundreds, thousands, or even tens-of-thousands of reviews can be analyzed. We will use the Cloud Natural Language API to measure its effectiveness in detecting positive and negative jargon and emotions from real customer review of popular Amazon products.

In [ ]:
import getpass
APIKEY = getpass.getpass()

In [ ]:
APIKEY

In [ ]:
# import build
from googleapiclient.discovery import build

# Methodology

Our approach to our research involved looking at products on Amazon that were considered relatively popular in their categories of clothing, shoes, and electronics. Many of the products' reviews analyzed were "Amazon's Choice" products that are at the top of the search page. We looked at top reviews for a product (4 or 5 stars) as well as negative reviews. They were compiled into a spreadsheet (linked below) that contains factors such as the review, whether it was good or bad, the overall rating, number of ratings, and calculated values from our the API such as average magnitude.


[Raw Data in Google Spreadsheet](https://docs.google.com/spreadsheets/d/1VXJOkV1LXx0ZD6oOg-_DpD6E0cWMeh2T5dD7-6H52VU/edit#gid=879487987)

# Data Table

In [46]:
from google.colab import auth
auth.authenticate_user()

import gspread
from google.auth import default
creds, _ = default()

gc = gspread.authorize(creds)

worksheet = gc.open('QTM 250 HW 4').sheet1

# get_all_values gives a list of rows.
rows = worksheet.get_all_values()
print(rows)

# Convert to a DataFrame and render.
import pandas as pd
pd.DataFrame.from_records(rows)

[['Product Type', 'Data #', 'Quotation', 'Average Polarity', 'Average Magnitude', 'Review Number', 'Link to Product', 'Overall Rating for Product (out of 5)', 'Rating (stars)', 'Name for Who Found the Data'], ['Clothing', 'Product #1 - Good', "   'This top is so pretty, absolutely LOVE the petal sleeves!', 'This top is like a t-shirt but with fancier sleeves, which makes it easy to dress up but you still get the comfort of a tee.', 'The material is very soft, fairly lightweight & loose fit.','I’m a big fan of v-neck shirts, I think they are more feminine.', 'The v-neckline combined with the petal sleeves makes this top very flattering & pretty.','True to size & color is accurate to stock photo.', 'Overall, I am very happy with this purchase.', 'Great product, good condition & fast shipping!', 'If you found my review helpful, I would appreciate you clicking “helpful” - thanks!!!'", '1', '0.83', '12,392', 'https://www.amazon.com/Shirts-Women-Summer-Sleeve-Black/dp/B08S34FZKM/ref=zg-bs_fa

0                  1  \
0   Product Type             Data #   
1       Clothing  Product #1 - Good   
2       Clothing   Product #1 - Bad   
3       Clothing  Product #2 - Good   
4       Clothing   Product #2 - Bad   
5       Clothing  Product #3 - Good   
6       Clothing   Product #3 - Bad   
7          Shoes   Product #4- Good   
8          Shoes    Product #4- Bad   
9          Shoes   Product #5- Good   
10         Shoes    Product #5- Bad   
11         Shoes  Product #6 - Good   
12         Shoes    Product #6- Bad   
13   Electronics   Product #7- Good   
14   Electronics    Product #7- Bad   
15   Electronics   Product #8- Good   
16   Electronics    Product #8- Bad   
17   Electronics   Product #9- Good   
18   Electronics    Product #9- Bad   

                                                    2                 3  \
0                                           Quotation  Average Polarity   
1      'This top is so pretty, absolutely LOVE the...                 1   
2   Fabric quality is horrendous and varies from s...                -1   
3   My favorite shirts I’ve been searching for mor...                 1   
4   The fit was ok...but the collars are terrible....                -1   
5   very cute, and very comfortable, was very plea...                 1   
6   The delivery went well, but the size is way of...                -1   
7   Loved the style and the comfort.', 'I would bu...                 1   
8   They are nothing like what is pictured.', 'col...                -1   
9   No tired feet at the end of the day.', 'New fa...                 1   
10  bottom of the shoe is already tearing away.', ...                -1   
11  They are so comfortable.', 'I’ve received so m...                 1   
12  Product very loose in the shoebox. ', 'Just so...                -1   
13  I'm soooooo happy with this amazing device!!!!...                 1   
14  Don't waste your money this product is awful. ...                -1   
15  This USB C Hub comes with a great quality for ...                 1   
16  I was truly disappointed and embarrassed. I wi...                -1   
17  This mouse is very nice! I am a nurse in a lon...                 1   
18  It is exactly what it said it was, but I was v...                -1   

                    4              5  \
0   Average Magnitude  Review Number   
1                0.83         12,392   
2                 0.7         12,392   
3                0.82          10033   
4                0.65          10033   
5                 0.9            469   
6                0.53            469   
7                1.04         40,359   
8                 0.8         40,359   
9                 0.9          9,804   
10                0.4          9,804   
11                0.9         29,430   
12               0.65         29,430   
13                0.9         93,949   
14               0.66         93,949   
15                0.9          32448   
16               0.85         32,448   
17               0.87          37859   
18                0.7          37859   

                                                    6  \
0                                     Link to Product   
1   https://www.amazon.com/Shirts-Women-Summer-Sle...   
2   https://www.amazon.com/Shirts-Women-Summer-Sle...   
3   https://www.amazon.com/Polo-Ralph-Lauren-Class...   
4   https://www.amazon.com/Polo-Ralph-Lauren-Class...   
5   https://www.amazon.com/Summer-Tshirt-Casual-Tr...   
6   https://www.amazon.com/Summer-Tshirt-Casual-Tr...   
7   https://www.amazon.com/Hey-Dude-Womens-Loafer-...   
8   https://www.amazon.com/Hey-Dude-Womens-Loafer-...   
9   https://www.amazon.com/Running-Lightweight-Cus...   
10  https://www.amazon.com/Running-Lightweight-Cus...   
11  https://www.amazon.com/dp/B0878SLRFN/ref=sspa_...   
12  https://www.amazon.com/dp/B0878SLRFN/ref=sspa_...   
13  https://www.amazon.com/Ring-Video-Doorbell-Sat...   
14  https://www.amazon.com/Ring-Video-Doorbell-Sat...   
15  https://www.amaz

# Github Repository Link

[GitHub Repository](https://github.com/ayeshasaeed97/qtm-250-hw-4-group-5)

Good review: 4 & 5 out of 5

Bad review: 1,2,3 out of 5

3 is medium, but normally, people are lenient about giving good rating to products. We will consider that 3/5 is a "bad" review

We used words with obvious positive and negative connotations to see how the API would rate them. As we can see, the API recognizes the words as we would expect them. This allows us to establish a benchmark for the API in future usage.

In [47]:
lservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
  'Good',
  'Bad',
  'Great',
  'Awful',
  'Amazing',
  'Horrible',
  'The'
]
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

POLARITY=1 MAGNITUDE=0.8 for Good
POLARITY=1 MAGNITUDE=0.3 for Bad
POLARITY=1 MAGNITUDE=0.9 for Great
POLARITY=-1 MAGNITUDE=0.8 for Awful
POLARITY=1 MAGNITUDE=0.9 for Amazing
POLARITY=1 MAGNITUDE=0.3 for Horrible
POLARITY=1 MAGNITUDE=0.2 for The


#Product 1

https://raw.githubusercontent.com/hasongcho/QTM250-GroupProj/main/Screen%20Shot%202022-04-20%20at%203.19.54%20PM.png


Product Type: Clothing

Total Review Number: 12,392 reviews

Overall Rating: 4.4/5



## Good Review

Rating: 5/5

In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'This top is so pretty, absolutely LOVE the petal sleeves!', 'This top is like a t-shirt but with fancier sleeves, which makes it easy to dress up but you still get the comfort of a tee.', 'The material is very soft, fairly lightweight & loose fit.','I’m a big fan of v-neck shirts, I think they are more feminine.', 'The v-neckline combined with the petal sleeves makes this top very flattering & pretty.','True to size & color is accurate to stock photo.', 'Overall, I am very happy with this purchase.', 'Great product, good condition & fast shipping!', 'If you found my review helpful, I would appreciate you clicking “helpful” - thanks!!!']
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

POLARITY=1 MAGNITUDE=0.9 for This top is so pretty, absolutely LOVE the petal sleeves!
POLARITY=1 MAGNITUDE=0.9 for This top is like a t-shirt but with fancier sleeves, which makes it easy to dress up but you still get the comfort of a tee.
POLARITY=1 MAGNITUDE=0.8 for The material is very soft, fairly lightweight & loose fit.
POLARITY=1 MAGNITUDE=0.9 for I’m a big fan of v-neck shirts, I think they are more feminine.
POLARITY=1 MAGNITUDE=0.9 for The v-neckline combined with the petal sleeves makes this top very flattering & pretty.
POLARITY=1 MAGNITUDE=0.9 for True to size & color is accurate to stock photo.
POLARITY=1 MAGNITUDE=0.9 for Overall, I am very happy with this purchase.
POLARITY=1 MAGNITUDE=0.9 for Great product, good condition & fast shipping!
POLARITY=1 MAGNITUDE=0.4 for If you found my review helpful, I would appreciate you clicking “helpful” - thanks!!!


##Negative Review

Rating 1/5

In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'Fabric quality is horrendous and varies from shirt to shirt.', 'I had to reorder due to the first top being frayed and in questionable condition, but the second top that arrived was a completely different fabric that was too shiny to be a casual black T-shirt type of material.', 'In a word, tacky.' ,'The bag it was enclosed in looked like it was reused multiple times from past returns.', 'Such a hassle to order and return items that do not match the photos or vendor’s description.','The style would be cute to wear if the manufacturer or vendor cared about quality and consistency.']
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

POLARITY=-1 MAGNITUDE=0.8 for Fabric quality is horrendous and varies from shirt to shirt.
POLARITY=-1 MAGNITUDE=0.6 for I had to reorder due to the first top being frayed and in questionable condition, but the second top that arrived was a completely different fabric that was too shiny to be a casual black T-shirt type of material.
POLARITY=-1 MAGNITUDE=0.9 for In a word, tacky.
POLARITY=-1 MAGNITUDE=0.7 for The bag it was enclosed in looked like it was reused multiple times from past returns.
POLARITY=-1 MAGNITUDE=0.8 for Such a hassle to order and return items that do not match the photos or vendor’s description.
POLARITY=-1 MAGNITUDE=0.4 for The style would be cute to wear if the manufacturer or vendor cared about quality and consistency.


# Product #2

https://raw.githubusercontent.com/hasongcho/QTM250-GroupProj/main/Clothing%232.png

Product Type: Clothing

Total Review Number: 10,033 reviews

Overall Rating: 4.6/5

##Good Review 

Rating: 5/5


In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'My favorite shirts I’ve been searching for more polo shirts, but a variety pack and my criteria was basically a light blue, darker shade of blue, and another black shirt.', 'To find this pack felt awesome haha I’m very happy to have found these!', 'You’ll never regret buying these t shirts they last forever, keep its shape very nice, over the course of time becomes significantly more comfortable, and somehow these t shirts just look sharp.', 'I can’t really describe it, but Calvin Klein also does a good job doing this.', 'Not quite sure how they do it, but I’ll never complain about that!']
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

##Negative Review

Rating 1/5


In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'The fit was ok...but the collars are terrible.', 'They are bunched up and look like a $2 shirt from Walmart.']
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

POLARITY=-1 MAGNITUDE=0.6 for The fit was ok...but the collars are terrible.
POLARITY=-1 MAGNITUDE=0.7 for They are bunched up and look like a $2 shirt from Walmart.


# Product #3

https://raw.githubusercontent.com/hasongcho/QTM250-GroupProj/main/Clothing%20%233.png

Product Type: Clothing

Total Review Number: 469 ratings

Overall Rating: 3.2/5


##Good Review

Rating: 5/5

In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'very cute, and very comfortable, was very pleased']
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

POLARITY=1 MAGNITUDE=0.9 for very cute, and very comfortable, was very pleased


##Negative Review

Rating 3/5


In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'The delivery went well, but the size is way off.', 'I usually wear and XL, so I ordered that size.', 'Well it was very tight.', 'I would of needed to order two more sizes for it to fit me.', 'I also did not like the feel of the fabric.', 'I would not recommend this product.'
]
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

POLARITY=-1 MAGNITUDE=0.6 for The delivery went well, but the size is way off.
POLARITY=1 MAGNITUDE=0.2 for I usually wear and XL, so I ordered that size.
POLARITY=-1 MAGNITUDE=0.5 for Well it was very tight.
POLARITY=-1 MAGNITUDE=0.2 for I would of needed to order two more sizes for it to fit me.
POLARITY=-1 MAGNITUDE=0.9 for I also did not like the feel of the fabric.
POLARITY=-1 MAGNITUDE=0.8 for I would not recommend this product.


##Product  #4
https://github.com/juliezhang1/shoes/blob/b5886185cf8cc1d5f03fd6597d834840a20ce995/Screen%20Shot%202022-04-20%20at%208.47.33%20PM.png


Product Type: Shoes

Total Review Number: 40,359 ratings

Overall Rating: 4.7/5

##Positive Review

Rating: 5/5

In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'Loved the style and the comfort.', 'I would buy again.', 'The sole is fabulous.', 'They fit like a glove.', 'Superior comfort.', 'Love these! Glitter is beautiful.', 'These are so cute.']
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

##Negative review
Rating: 1/5

In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'They are nothing like what is pictured.', 'color & style are way off so I went to return them.', 'fit was horrible.'
]
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

##Product #5

https://github.com/juliezhang1/shoes/blob/83f2bffb545b53f765c1c82668d19e25bc27f760/Screen%20Shot%202022-04-20%20at%208.52.59%20PM.png

Product Type: Shoes

Total Review Number: 9,804 ratings

Overall Rating: 4.3/5

##Positive Reviews 
Rating: 5/5

In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'No tired feet at the end of the day.', 'New favorite shoes.', 'Will buy more.']
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

##Negative Reviews 
Rating: 2/5

In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'bottom of the shoe is already tearing away.', 'Less than 3 months of wear.']
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

##Product #6

https://github.com/juliezhang1/shoes/blob/7a7d69dc9175a9e3f27fda8e3a8b603648edc111/Screen%20Shot%202022-04-20%20at%208.53.06%20PM.png

Product Type: Shoes

Total Review Number: 29,430 ratings

Overall Rating: 4.3/5

##Positive Reviews
Rating: 5/5

In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'They are so comfortable.', 'I’ve received so many compliments.']
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

#Negative Reviews
Rating:1/5

In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'Product very loose in the shoebox. ', 'Just so you know, they cost triple to send back if they don’t fit!']
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

#Product 7

Electronics

4.7/5

Total Product Reviews: 93,949

## Positive Review

In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'Im soooooo happy with this amazing device!!!!', 'I want to give all my friends and family ones as a gifts and I want to buy one for my back door too!', 'This has been the best 100$ investment this year-well worth it!!!', 'I wish I would have bought one sooner.', 'Also, the neighbors app....amazing and I cant stop checking it out.'
]
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

POLARITY=1 MAGNITUDE=0.9 for Im soooooo happy with this amazing device!!!!
POLARITY=1 MAGNITUDE=0.9 for I want to give all my friends and family ones as a gifts and I want to buy one for my back door too!
POLARITY=1 MAGNITUDE=0.9 for This has been the best 100$ investment this year-well worth it!!!
POLARITY=1 MAGNITUDE=0.9 for I wish I would have bought one sooner.
POLARITY=1 MAGNITUDE=0.9 for Also, the neighbors app....amazing and I cant stop checking it out.


## Negative Review

In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'Dont waste your money this product is awful.', 'It doesnt catch motion of a human until someone is right up on it and then its too late for the image to be usefulvacated it will catch their ear as they walk by.', 'The mail man it will catch pull up but vacated ur takes him a second to get out the car or stops recording veggie he can get to it and it catches every single car that goes by my house no matter how I change the settings and detection around.', 'It is literal garbage in ever way and not useful for any application.', 'The notification it seems to my phone is also delayed and it is getting good signal from our internet.', 'It takes all day to recharge it as well.', 'This was a total waste of money!'
]
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

POLARITY=-1 MAGNITUDE=0.7 for Dont waste your money this product is awful.
POLARITY=-1 MAGNITUDE=0.7 for It doesnt catch motion of a human until someone is right up on it and then its too late for the image to be usefulvacated it will catch their ear as they walk by.
POLARITY=1 MAGNITUDE=0.2 for The mail man it will catch pull up but vacated ur takes him a second to get out the car or stops recording veggie he can get to it and it catches every single car that goes by my house no matter how I change the settings and detection around.
POLARITY=-1 MAGNITUDE=0.8 for It is literal garbage in ever way and not useful for any application.
POLARITY=-1 MAGNITUDE=0.7 for The notification it seems to my phone is also delayed and it is getting good signal from our internet.
POLARITY=-1 MAGNITUDE=0.7 for It takes all day to recharge it as well.
POLARITY=-1 MAGNITUDE=0.8 for This was a total waste of money!


# Product 8

Electronics

4.6/5

Overall Product Reviews: 32,448

## Positive Review

In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'This USB C Hub comes with a great quality for a great price.', 'Definitely worth my money.', 'It is light weighted and small size, so it is easy for traveling.' ,'I wish I can get this earlier.', 'This hub is a pretty cool design!', 'I would highly recommend this to people who need a hub like this and dont want to spend a lot of money.'
]
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

POLARITY=1 MAGNITUDE=0.9 for This USB C Hub comes with a great quality for a great price.
POLARITY=1 MAGNITUDE=0.9 for Definitely worth my money.
POLARITY=1 MAGNITUDE=0.9 for It is light weighted and small size, so it is easy for traveling.
POLARITY=1 MAGNITUDE=0.8 for I wish I can get this earlier.
POLARITY=1 MAGNITUDE=0.9 for This hub is a pretty cool design!
POLARITY=1 MAGNITUDE=0.9 for I would highly recommend this to people who need a hub like this and dont want to spend a lot of money.


## Negative Review

In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'I was truly disappointed and embarrassed.' ,'I will be returning this product, and buying something with better reviews and hopefully works without getting so hot.'
]
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

POLARITY=-1 MAGNITUDE=0.9 for I was truly disappointed and embarrassed.
POLARITY=-1 MAGNITUDE=0.8 for I will be returning this product, and buying something with better reviews and hopefully works without getting so hot.


# Product 9

Electronics

4.4/5

Total Product Reviews: 37859


## Positive Review

In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'This mouse is very nice!', 'I am a nurse in a long term care facility and we never use the word “quiet” but the clicks on here are nonexistent.' ,'It’s nice to not have to hear hundreds of clicks because it’s annoying after awhile, this is an awesome perk for me.'
]
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

POLARITY=1 MAGNITUDE=0.9 for This mouse is very nice!
POLARITY=-1 MAGNITUDE=0.8 for I am a nurse in a long term care facility and we never use the word “quiet” but the clicks on here are nonexistent.
POLARITY=1 MAGNITUDE=0.9 for It’s nice to not have to hear hundreds of clicks because it’s annoying after awhile, this is an awesome perk for me.


## Negative Review

In [ ]:
wlservice = build('language', 'v1beta1', developerKey=APIKEY)
quotes = [
      'It is exactly what it said it was, but I was very disappointed to find out that I cant change what the buttons do' ,'The provided poorly-edited, single-fold 6"x4" scrap of paper called "Quick User Manual" is obviously meant to apply to dozens of different models and gives no details beyond setup.'
]
for quote in quotes:
  response = lservice.documents().analyzeSentiment(
    body={
      'document': {
         'type': 'PLAIN_TEXT',
         'content': quote
      }
    }).execute()
  polarity = response['documentSentiment']['polarity']
  magnitude = response['documentSentiment']['magnitude']
  print('POLARITY=%s MAGNITUDE=%s for %s' % (polarity, magnitude, quote))

POLARITY=-1 MAGNITUDE=0.6 for It is exactly what it said it was, but I was very disappointed to find out that I cant change what the buttons do
POLARITY=-1 MAGNITUDE=0.8 for The provided poorly-edited, single-fold 6"x4" scrap of paper called "Quick User Manual" is obviously meant to apply to dozens of different models and gives no details beyond setup.


# Analysis
![Graph 1](https://github.com/ayeshasaeed97/QTM250-example/blob/main/Screen%20Shot%202022-04-20%20at%2010.09.18%20PM.png?raw=true)

This visualization was made to see how a product's magnitude changes depending on whether we analyzed the positive or negative review. As we can see, a strong magnitude for a positive review is returned with a similar, if not slightly weaker, magnitude for a negative review. This shows that people might be using more emotionally charged words when satisfied with a product versus when not. However, we must acknowledge that we generally chose well-rated products overall. Perhaps if we chose a worse-rated product, that reviews would have more negatively charged wording.


![Graph 2](https://github.com/ayeshasaeed97/QTM250-example/blob/main/Screen%20Shot%202022-04-20%20at%2010.09.36%20PM.png?raw=true)

This visualization shows us that positive reviews are overal +1 in polarity while negative reviews are -1. We can see that the magnitude of reviews is higher for positive reviews as opposed to negative reviews. Once again, this could be due to the researcher's pick of reviews rather than being an accurate respresentation of 

![Graph 3](https://github.com/ayeshasaeed97/QTM250-example/blob/main/Screen%20Shot%202022-04-20%20at%2010.09.47%20PM.png?raw=true)

This visualization is interesting as it tells us that, generally, products with more reviews have higher magnitudes. This could be that the more people like or dislike a product, the more likely they will go out of their way to write a review. They likely have strong opinions to write a review which results in their wording being more charged one way or another.


# Critiques of the API

Overall, this API was easy and enjoyable to use. However, I realized that there are limitations in its output. For example, the API only gives a -1 or 1 for polarity. It does not really lable a word as neutral, like a value of 0, as we can see in our practice API call with the word "The" where it gave it a +1 polarity, but a very small magnitude. "The" is a very neutral word, so we wonder where the API derieves how it will judge a word.

#Architecture

![Arch](https://github.com/ayeshasaeed97/qtm-250-hw-4-group-5/blob/main/Screen%20Shot%202022-04-20%20at%2011.09.30%20PM.png?raw=true)
